week_4/invert_func_with_multiplication_division/invert_func_with_multiplication_division.cpp

```diff
@@ -0,0 +1,103 @@
+#include <algorithm>
+#include <iostream>
+#include <vector>
+
+struct Image {
+  double quality;
+  double freshness;
+  double rating;
+};
+
+struct Params {
+  double a;
+  double b;
+  double c;
+};
+
+
+class FunctionPart {
+ 
+  private:
+  char operation;
+  double value;
+ 
+ public:
+  FunctionPart(char new_operation, double new_value) {
+    value = new_value;
+    operation = new_operation;
+  }
+  double Apply(const double& sourse_value) const {
+    if (operation == '+') {
+      return sourse_value + value;
+    } else if (operation == '-') {
+      return sourse_value - value;
+    } else if (operation == '*') {
+      return sourse_value * value;
+    } else {
+      return sourse_value / value;
+    }
+  }
+  void Invert() {
+    if (operation == '+') {
+      operation = '-';
+    } else if (operation == '-') {
+      operation = '+';
+    } else if (operation == '*') {
+      operation = '/';
+    } else {
+      operation = '*';
+    }
+  }
+ };
+
+class Function {
+
+private:
+  std::vector<FunctionPart> parts;
+  
```

Comment:
@Ypostolov What is the constructor for Function?

Reply:
Please clarify.